rest_framework/templatetags/rest_framework.py

```diff
@@ -135,11 +135,16 @@ def optional_logout(request, user):
             <b class="caret"></b>
         </a>
         <ul class="dropdown-menu">
-            <li><a href='{href}?next={next}'>Log out</a></li>
+            <form id="logoutForm" method="post" action="{href}?next={next}">
+                <input type="hidden" name="csrfmiddlewaretoken" value="{csrf_token}">
+            </form>
+            <li>
+                <a href="#" onclick='document.getElementById("logoutForm").submit()'>Log out</a>
```

Comment:
not sure, but should this href use #?

Reply:
Since it immediately redirects the user to `next`, it should not be a problem?

Alternatives:

* We could remove the href. In that case, we would need to add inline css so that cursor would be pointer. Also we'll probably need to add `role="button"` for improved semantics for screen readers.

* We could set `next` as href. Although I'm not sure how it would behave. We may need to call `preventDefault`.

I'm refraining from using `<button>` here since it would require a bit of CSS juggling.

Reply:
It's a pretty standard pattern I think, seems okay to me.